File: app/src/main/java/ru/evotor/framework/core/action/event/cash_drawer/CashDrawerEvent.java
package ru.evotor.framework.core.action.event.cash_drawer;

import android.os.Bundle;

public abstract class CashDrawerEvent {
    private static final String KEY_CASH_DRAWER_ID = "cashDrawerId";

    private final int cashDrawerId;

    public CashDrawerEvent(Bundle extras) {
        this(
                extras.getInt(KEY_CASH_DRAWER_ID, -1)
        );
    }

    public CashDrawerEvent(int cashDrawerId) {
        this.cashDrawerId = cashDrawerId;
    }

    public Bundle toBundle() {
        Bundle result = new Bundle();
        result.putInt(KEY_CASH_DRAWER_ID, cashDrawerId);
        return result;
    }

    public int getKeyCashDrawerId() {
        return cashDrawerId;
    }
}

File: src/main/java/ru/evotor/framework/device/cash_drawer/event/CashDrawerEvent.kt
package ru.evotor.framework.device.cash_drawer.event

import android.os.Bundle

import ru.evotor.IBundlable

abstract class CashDrawerEvent internal constructor(val cashDrawerId: Int) : IBundlable {

    override fun toBundle(): Bundle {
        val result = Bundle()
        result.putInt(KEY_CASH_DRAWER_ID, cashDrawerId)
        return result
    }

    companion object {

        private const val KEY_CASH_DRAWER_ID = "cashDrawerId"

        internal fun getCashDrawerId(bundle: Bundle) = bundle.getInt(KEY_CASH_DRAWER_ID, -1)

    }
}
